### DISCUSSION TITLE
Can be transaction_guard_t rollback method noexcept?

### DISCUSSION THREAD

**Discussion by SergeyBakev** [fnc12/sqlite_orm#913] [Q&A]
Can be transaction_guard_t rollback method noexcept?

**Comment by fnc12** [comment]
Yes but what is your plan in case sqlite engine returns non `SQLITE_OK` result then?

**Comment by SergeyBakev** [comment]
std::terminate

**Reply by fnc12** [reply]
No no no, I mean how rollback will tell you that something went wrong?

**Comment by juandent** [comment]
Hi Eugene

I have been progressing slowly with the course on sqlite_orm
I was wondering if you have some uml diagrams that describe the general architecture of your library 

Thanks
Juan Dent

**Reply by fnc12** [reply]
Hi Juan. I don't have any. What is the purpose of UML diagram for `sqlite_orm`? I never faced a situation when someone required any UML diagram to understand how everything works.
